shell/platform/windows/flutter_windows_unittests.cc

```diff
@@ -38,6 +38,32 @@ TEST_F(WindowsTest, LaunchMain) {
   ASSERT_NE(controller, nullptr);
 }
 
+// Verify there is no unexpected output from launching main.
+TEST_F(WindowsTest, LaunchMainHasNoOutput) {
+  // Replace stdout & stderr stream buffers with our own.
+  std::stringstream cout_buffer;
+  std::stringstream cerr_buffer;
+  std::streambuf* old_cout_buffer = std::cout.rdbuf();
+  std::streambuf* old_cerr_buffer = std::cerr.rdbuf();
+  std::cout.rdbuf(cout_buffer.rdbuf());
+  std::cerr.rdbuf(cerr_buffer.rdbuf());
+
+  auto& context = GetContext();
+  WindowsConfigBuilder builder(context);
+  ViewControllerPtr controller{builder.Run()};
+  ASSERT_NE(controller, nullptr);
+
+  // Restore original stdout & stderr stream buffer.
+  std::cout.rdbuf(old_cout_buffer);
+  std::cerr.rdbuf(old_cerr_buffer);
+
+  // Verify stdout & stderr have no output.
+  std::string cout = cout_buffer.str();
+  std::string cerr = cerr_buffer.str();
+  EXPECT_TRUE(cout.empty());
+  EXPECT_TRUE(cerr.empty());
```

Comment:
You can use `testing::internal::CaptureStdout()` and `testing::internal::GetCapturedStdout()` to make this more succinct.

Reply:
I considered this but then found this message by the gtest folks: https://groups.google.com/g/googletestframework/c/7OTybZo-jp8

> First of all, you should not use APIs marked as "internal".
They are not part of the public API, are not supported and are subject to change at any time.

And:

> No, those APIs are internal for testing gTest itself.
They do so by overriding the stdout/stderr file descriptors and forwarding the data into a file, which means that they interfere with the actual output of the binary.
They could forward the output to the actual stdout/err, but that is more complex and there is no reason to do so on these internal APIs.
>
> It is rather trivial to capture std::cout/std::cerr (not stdout/stderr) on your side, if you want.
Eg: http://stackoverflow.com/questions/5419356/redirect-stdout-stderr-to-a-string

I think if this becomes a common pattern we could introduce a helper to prevent possible gotchas.

Reply:
Oops I added the `autosubmit` to this before letting you reply. Let me know if you have feedback, I'd be happy to send a follow-up pull request. Sorry about that!

Reply:
Seems reasonable. If we find ourselves needing to do this more than once we should create an RAII wrapper that does the grabbing and restoring.